src/obstacle_detection.cpp

```diff
@@ -1,43 +0,0 @@
-#include "geometry_msgs/Point.h"
-#include "obstacle.h"
-#include "obstacletype.h"
-#include <vector>
-
-// LaserScan
-float angle_min; // minimum angle possible
-float angle_max; // maximum angle possible
-float angle_increment; // difference in angle for each reading
-float range_min; // minimum value of ranges
-float range_max; // maximum value of ranges
-float* ranges; // list of distance 'r' from each measured point
```

Comment:
Dont ever do this. Most segfaults are caused by indexing issues. 
For arrays use std::array E.g.
std::array ranges;
to get/set values use at()
ranges.at(0) = 5;
int var = ranges.at(3);